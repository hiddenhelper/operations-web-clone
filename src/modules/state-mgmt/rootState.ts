import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
