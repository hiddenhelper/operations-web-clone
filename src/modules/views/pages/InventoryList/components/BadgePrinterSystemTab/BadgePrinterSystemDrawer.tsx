import React, { memo } from 'react';

import BadgePrintingSystemSummary from '../../../../shared/BadgePrintingSystemSummary';
import DeviceDrawer from '../DeviceDrawer';

import { GeneralModel, BadgePrintingSystemModel } from '../../../../../models';

export interface IBadgePrintingSystemDrawerProps {
  device: BadgePrintingSystemModel.IBadgePrintingSystem;
  deviceListElement: React.ReactNode;
  isOpen: boolean;
  isLoading: boolean;
  deleteLoading: GeneralModel.ILoadingStatus;
  onClose: () => void;
  onDelete: (id: string) => void;
}

const BadgePrinterSystemDrawer = ({ isOpen, isLoading, deleteLoading, deviceListElement, device, onClose, onDelete }: IBadgePrintingSystemDrawerProps) => {
  return (
    <>
      <DeviceDrawer
        title="BPS Information"
        dataTestId="bps-drawer-detail"
        nameField="name"
        deviceKey="badge-printing-system"
        deleteBtnTestId="deleteBpsBtn"
        isOpen={isOpen}
        isLoading={isLoading}
        onClose={onClose}
        device={device}
        deviceListElement={deviceListElement}
        deleteLoading={deleteLoading}
        type="BPS"
        onDelete={onDelete}
        renderSummary={() => <BadgePrintingSystemSummary device={device} deviceListElement={deviceListElement} />}
      />
    </>
  );
};

export default memo(BadgePrinterSystemDrawer);
