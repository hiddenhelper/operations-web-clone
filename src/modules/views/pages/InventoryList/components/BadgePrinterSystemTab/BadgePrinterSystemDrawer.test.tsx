import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import BadgePrinterSystemDrawer, { IBadgePrintingSystemDrawerProps } from './BadgePrinterSystemDrawer';
import { getBadgePrinterSystem_1, getBadgePrinterSystem_2 } from '../../../../../../test/entities';
import { createMockStore } from 'redux-test-utils';
import { getInitialState } from '../../../../../../test/rootState';
import { Provider } from 'react-redux';

describe('BadgePrinterSystemDrawer', () => {
  let props: IBadgePrintingSystemDrawerProps;

  beforeEach(() => {
    props = {
      isOpen: true,
      isLoading: false,
      deleteLoading: undefined,
      deviceListElement: null,
      device: getBadgePrinterSystem_1(),
      onClose: jest.fn(),
      onDelete: jest.fn(),
    };
  });

  it('should render', () => {
    const { container } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <BadgePrinterSystemDrawer {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should show loading', () => {
    props.isLoading = true;
    props.deleteLoading = { isLoading: false, error: null, hasError: false };
    const wrapper = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <BadgePrinterSystemDrawer {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.container).toMatchSnapshot();
    expect(wrapper.getByText('Loading...'));
  });

  it('should render alternative values', () => {
    props.device = getBadgePrinterSystem_2();
    const { container } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <BadgePrinterSystemDrawer {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
