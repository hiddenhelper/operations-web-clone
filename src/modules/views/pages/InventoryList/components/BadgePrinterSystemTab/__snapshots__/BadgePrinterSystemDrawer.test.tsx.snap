// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgePrinterSystemDrawer should render 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-82 open"
    data-testid="bps-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-83 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="makeStyles-drawerButtonWrapper-85"
      >
        <div
          class="makeStyles-drawerButton-86"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-84"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-87 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            BPS Information
          </p>
        </div>
      </div>
      <div
        class="makeStyles-listDetail-72"
      >
        <div
          class="makeStyles-summaryDeviceDetail-99"
        >
          <div
            class="makeStyles-company-103"
          >
            <div
              class="makeStyles-drawerOverflowWrapper-118"
            >
              <div
                class="makeStyles-drawerText-110"
              >
                <p
                  class="MuiTypography-root makeStyles-deviceAccent-104 makeStyles-entityTitle-113 MuiTypography-body1"
                >
                  Badge Printer Name
                </p>
                <div
                  class="makeStyles-drawerMainText-111"
                >
                  <span>
                    No project assigned
                  </span>
                  <span>
                    Available
                  </span>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <svg
                  class="makeStyles-drawerNotesIcon-107"
                >
                  notes.svg
                </svg>
                <div
                  class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                >
                  <span>
                    some notes
                  </span>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerText-110"
                >
                  <p
                    class="MuiTypography-root makeStyles-deviceAccent-104 makeStyles-entityTitle-113 MuiTypography-body1"
                  >
                    Laptop
                  </p>
                  <div
                    class="makeStyles-drawerMainText-111"
                  >
                    <span>
                      AFF12903112
                    </span>
                    <span>
                      thinkpad x1
                    </span>
                    <span>
                      18.0.1
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerColumnDirection-117"
                >
                  <div
                    class="makeStyles-drawerRowDirection-116"
                  >
                    <svg
                      class="makeStyles-drawerMonetizationIcon-109 makeStyles-bpsIconSize-119"
                    >
                      monetization.svg
                    </svg>
                    <div
                      class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                    >
                      <span>
                        $
                        3200
                      </span>
                      <span>
                        Lenovo
                      </span>
                      <span>
                        Invoiced 
                        invoice number
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerRowDirection-116"
                >
                  <svg
                    class="makeStyles-bpsIconSize-119"
                  >
                    device-status.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                  >
                    <span>
                      Order/Purchase Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Warranty Expiration Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      In Service Date:
                       
                      May 19, 2020
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerRowDirection-116"
                >
                  <svg
                    class="makeStyles-drawerSecondaryNotesIcon-108"
                  >
                    notes.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                  >
                    <span>
                      notes
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerText-110"
                >
                  <p
                    class="MuiTypography-root makeStyles-deviceAccent-104 makeStyles-entityTitle-113 MuiTypography-body1"
                  >
                    Printer
                  </p>
                  <div
                    class="makeStyles-drawerMainText-111"
                  >
                    <span>
                      Zebra
                    </span>
                    <span>
                      12904328913
                    </span>
                    <span>
                      model type
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerColumnDirection-117"
                >
                  <div
                    class="makeStyles-drawerRowDirection-116"
                  >
                    <svg
                      class="makeStyles-drawerMonetizationIcon-109"
                    >
                      monetization.svg
                    </svg>
                    <div
                      class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                    >
                      <span>
                        $
                        300
                      </span>
                      <span>
                        vendor
                      </span>
                      <span>
                        Invoiced 
                        invoice
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerRowDirection-116"
                >
                  <svg
                    class="makeStyles-bpsIconSize-119"
                  >
                    device-status.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                  >
                    <span>
                      Order/Purchase Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Warranty Expiration Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Last Maintenance Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      In Service Date:
                       
                      May 19, 2020
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerRowDirection-116"
                >
                  <svg
                    class="makeStyles-drawerSecondaryNotesIcon-108"
                  >
                    notes.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                  >
                    <span>
                      some note
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerText-110"
                >
                  <p
                    class="MuiTypography-root makeStyles-deviceAccent-104 makeStyles-entityTitle-113 MuiTypography-body1"
                  >
                    Scanner
                  </p>
                  <div
                    class="makeStyles-drawerMainText-111"
                  >
                    <span>
                      BBF19021
                    </span>
                    <span>
                      model
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerColumnDirection-117"
                >
                  <div
                    class="makeStyles-drawerRowDirection-116"
                  >
                    <svg
                      class="makeStyles-drawerMonetizationIcon-109"
                    >
                      monetization.svg
                    </svg>
                    <div
                      class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                    >
                      <span>
                        $
                        1000
                      </span>
                      <span>
                        vendor
                      </span>
                      <span>
                        Invoiced 
                        invoice
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <div
                  class="makeStyles-drawerRowDirection-116"
                >
                  <svg
                    class="makeStyles-bpsIconSize-119"
                  >
                    device-status.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                  >
                    <span>
                      Order/Purchase Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Warranty Expiration Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      In Service Date:
                       
                      May 19, 2020
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-106"
              >
                <svg
                  class="makeStyles-drawerSecondaryNotesIcon-108"
                >
                  notes.svg
                </svg>
                <div
                  class="makeStyles-drawerText-110 makeStyles-drawerTextSeparation-112"
                >
                  <span>
                    -
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-drawerDivider-4"
        />
        <div
          class="makeStyles-ctaWrapper-75"
        >
          <a
            href="/inventory/badge-printing-system/wizard/77afbfce-1bca-31c3-b512-3a1ed78742bb"
          >
            <span
              class=""
              data-testid="controlled-button"
              tabindex="-1"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-41 makeStyles-borderPrimaryButton-33"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Edit
                </span>
              </button>
            </span>
          </a>
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-41 makeStyles-warningButton-42"
              data-testid="deleteBpsBtn"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Delete
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BadgePrinterSystemDrawer should render alternative values 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-299 open"
    data-testid="bps-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-300 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="makeStyles-drawerButtonWrapper-302"
      >
        <div
          class="makeStyles-drawerButton-303"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-301"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-304 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            BPS Information
          </p>
        </div>
      </div>
      <div
        class="makeStyles-listDetail-289"
      >
        <div
          class="makeStyles-summaryDeviceDetail-316"
        >
          <div
            class="makeStyles-company-320"
          >
            <div
              class="makeStyles-drawerOverflowWrapper-335"
            >
              <div
                class="makeStyles-drawerText-327"
              >
                <p
                  class="MuiTypography-root makeStyles-deviceAccent-321 makeStyles-entityTitle-330 MuiTypography-body1"
                >
                  Badge Printer Name 2
                </p>
                <div
                  class="makeStyles-drawerMainText-328"
                >
                  <span>
                    No project assigned
                  </span>
                  <span>
                    Assigned
                  </span>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <svg
                  class="makeStyles-drawerNotesIcon-324"
                >
                  notes.svg
                </svg>
                <div
                  class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                >
                  <span>
                    notes 2
                  </span>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerText-327"
                >
                  <p
                    class="MuiTypography-root makeStyles-deviceAccent-321 makeStyles-entityTitle-330 MuiTypography-body1"
                  >
                    Laptop
                  </p>
                  <div
                    class="makeStyles-drawerMainText-328"
                  >
                    <span>
                      AFF12903112
                    </span>
                    <span>
                      thinkpad x1
                    </span>
                    <span>
                      18.0.1
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerColumnDirection-334"
                >
                  <div
                    class="makeStyles-drawerRowDirection-333"
                  >
                    <svg
                      class="makeStyles-drawerMonetizationIcon-326 makeStyles-bpsIconSize-336"
                    >
                      monetization.svg
                    </svg>
                    <div
                      class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                    >
                      <span>
                        $
                        3200
                      </span>
                      <span>
                        Lenovo
                      </span>
                      <span>
                        Invoiced 
                        invoice number
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerRowDirection-333"
                >
                  <svg
                    class="makeStyles-bpsIconSize-336"
                  >
                    device-status.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                  >
                    <span>
                      Order/Purchase Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Warranty Expiration Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      In Service Date:
                       
                      May 19, 2020
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerRowDirection-333"
                >
                  <svg
                    class="makeStyles-drawerSecondaryNotesIcon-325"
                  >
                    notes.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                  >
                    <span>
                      notes
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerText-327"
                >
                  <p
                    class="MuiTypography-root makeStyles-deviceAccent-321 makeStyles-entityTitle-330 MuiTypography-body1"
                  >
                    Printer
                  </p>
                  <div
                    class="makeStyles-drawerMainText-328"
                  >
                    <span>
                      Data Card
                    </span>
                    <span>
                      332182912
                    </span>
                    <span>
                      model type
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerColumnDirection-334"
                >
                  <div
                    class="makeStyles-drawerRowDirection-333"
                  >
                    <svg
                      class="makeStyles-drawerMonetizationIcon-326"
                    >
                      monetization.svg
                    </svg>
                    <div
                      class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                    >
                      <span>
                        $
                        378
                      </span>
                      <span>
                        vendor 2
                      </span>
                      <span>
                        Invoiced 
                        invoice 2
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerRowDirection-333"
                >
                  <svg
                    class="makeStyles-bpsIconSize-336"
                  >
                    device-status.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                  >
                    <span>
                      Order/Purchase Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Warranty Expiration Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Last Maintenance Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      In Service Date:
                       
                      May 19, 2020
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerRowDirection-333"
                >
                  <svg
                    class="makeStyles-drawerSecondaryNotesIcon-325"
                  >
                    notes.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                  >
                    <span>
                      note printer 2
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerText-327"
                >
                  <p
                    class="MuiTypography-root makeStyles-deviceAccent-321 makeStyles-entityTitle-330 MuiTypography-body1"
                  >
                    Scanner
                  </p>
                  <div
                    class="makeStyles-drawerMainText-328"
                  >
                    <span>
                      BBF19021
                    </span>
                    <span>
                      model
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerColumnDirection-334"
                >
                  <div
                    class="makeStyles-drawerRowDirection-333"
                  >
                    <svg
                      class="makeStyles-drawerMonetizationIcon-326"
                    >
                      monetization.svg
                    </svg>
                    <div
                      class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                    >
                      <span>
                        $
                        1000
                      </span>
                      <span>
                        vendor
                      </span>
                      <span>
                        Invoiced 
                        invoice
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <div
                  class="makeStyles-drawerRowDirection-333"
                >
                  <svg
                    class="makeStyles-bpsIconSize-336"
                  >
                    device-status.svg
                  </svg>
                  <div
                    class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                  >
                    <span>
                      Order/Purchase Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      Warranty Expiration Date:
                       
                      May 19, 2020
                    </span>
                    <span>
                      In Service Date:
                       
                      May 19, 2020
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-drawerSection-323"
              >
                <svg
                  class="makeStyles-drawerSecondaryNotesIcon-325"
                >
                  notes.svg
                </svg>
                <div
                  class="makeStyles-drawerText-327 makeStyles-drawerTextSeparation-329"
                >
                  <span>
                    -
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-drawerDivider-221"
        />
        <div
          class="makeStyles-ctaWrapper-292"
        >
          <a
            href="/inventory/badge-printing-system/wizard/6631aacb-21fa-91cc-c5ff-55b12cbaf33b"
          >
            <span
              class=""
              data-testid="controlled-button"
              tabindex="-1"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-258 makeStyles-borderPrimaryButton-250"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Edit
                </span>
              </button>
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BadgePrinterSystemDrawer should show loading 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-201 open"
    data-testid="bps-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-202 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="makeStyles-drawerButtonWrapper-204"
      >
        <div
          class="makeStyles-drawerButton-205"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-203"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-206 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            BPS Information
          </p>
        </div>
      </div>
      <div>
        Loading...
      </div>
    </div>
  </div>
</div>
`;
